### PR TITLE
Opt-in to `ExperimentalSerializationApi` automatically

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: api/v1/mapping/build.gradle.kts
Comment (mnonnenmacher) [inline api/v1/mapping/build.gradle.kts:1]: Just to be sure, the compiler plugin is only required for code generation in modules that use the `@Serializable` annotation, but not in modules that just consume serializable classes, right?

File: build.gradle.kts [hunk 1]
```diff
@@ -164,9 +164,12 @@ subprojects {
     }
 
     tasks.withType<KotlinCompile>().configureEach {
+        val hasSerializationPlugin = plugins.hasPlugin(libs.plugins.kotlinxSerialization.get().pluginId)
+
         compilerOptions {
             allWarningsAsErrors = true
             jvmTarget = maxKotlinJvmTarget
+            if (hasSerializationPlugin) optIn = listOf("kotlinx.serialization.ExperimentalSerializationApi")
```
Comment (mnonnenmacher) [inline build.gradle.kts:172]: Could we just globally define this opt in, or would it fail in projects not depending on kotlinx.serialization?

File: api/v1/mapping/build.gradle.kts
Comment (sschuberth) [reply api/v1/mapping/build.gradle.kts:1]: Correct.

File: build.gradle.kts (hunk 1 above)
Comment (sschuberth) [reply build.gradle.kts:172]: It would fail.

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**